riposte-core/src/main/java/com/nike/riposte/client/asynchttp/netty/StreamingAsyncHttpClient.java

```diff
@@ -1139,28 +1155,65 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+            //      host/port we need for this call, then we can leave it alone and don't need to create a new one.
+            boolean requiresNewSslHandler = true;
+            if (registeredHandlerNames.contains(SSL_HANDLER_NAME)) {
+                SslHandler existingSslHandler = (SslHandler) p.get(SSL_HANDLER_NAME);
+                SSLEngine existingSslEngine = existingSslHandler.engine();
+                if (Objects.equals(downstreamHost, existingSslEngine.getPeerHost())
+                    && Objects.equals(downstreamPort, existingSslEngine.getPeerPort())
+                ) {
+                    // The existing SslHandler's SslEngine is pointed at the correct host/port. We don't need to
+                    //      add a new one.
+                    requiresNewSslHandler = false;
+                }
+                else {
+                    // The existing SslHandler's SslEngine is *not* pointed at the correct host/port. We need a new one,
+                    //      so remove the old one.
+                    p.remove(SSL_HANDLER_NAME);
+                }
+            }
+
+            if (requiresNewSslHandler) {
+                // SSL call and we need to add a SslHandler. Create the general-purpose reusable SslContexts if needed.
                 if (clientSslCtx == null) {
-                    if (relaxedHttpsValidation) {
-                        clientSslCtx =
-                            SslContextBuilder.forClient().trustManager(InsecureTrustManagerFactory.INSTANCE).build();
-                    }
-                    else {
-                        TrustManagerFactory tmf =
-                            TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
-                        tmf.init((KeyStore) null);
-                        clientSslCtx = SslContextBuilder.forClient().trustManager(tmf).build();
-                    }
+                    TrustManagerFactory tmf = TrustManagerFactory.getInstance(
+                        TrustManagerFactory.getDefaultAlgorithm()
+                    );
+                    tmf.init((KeyStore) null);
+
+                    clientSslCtx = SslContextBuilder
+                        .forClient()
+                        .trustManager(tmf)
+                        .build();
                 }
 
-                p.addAfter(DOWNSTREAM_CALL_TIMEOUT_HANDLER_NAME, SSL_HANDLER_NAME, clientSslCtx.newHandler(ch.alloc()));
+                if (insecureSslCtx == null) {
+                    insecureSslCtx = SslContextBuilder
+                        .forClient()
+                        .trustManager(InsecureTrustManagerFactory.INSTANCE)
+                        .build();
+                }
+
+                // Figure out which SslContext to use for this call.
+                SslContext sslCtxToUse = (relaxedHttpsValidation) ? insecureSslCtx : clientSslCtx;
+
+                // Create the SslHandler and configure the SslEngine
```

Comment:
Please put back the comment addendum: `//      as per the javadocs for SslContext.newHandler(ByteBufAllocator, String, int).`

I think it's important to know where we got the info from on how the SslEngine is supposed to be configured (in case it changes in the future and the javadocs for that method change, for example - then we'd know we need to adjust as well).

Reply:
done!